runtime/vm/vmthinit.c

```diff
@@ -80,6 +82,13 @@ UDATA initializeVMThreading(J9JavaVM *vm)
 		initializeMonitorTable(vm)
 	)
 	{
+#ifdef J9VM_GC_FINALIZATION
+		/* processReferenceMonitor is only used for Java 9 and later */
```

Comment:
Monitor should be initialized for Java 9 only. Destroying it for java 8 looks ugly